astropy/time/formats.py

```diff
@@ -619,8 +621,8 @@ def set_jds(self, val1, val2):
         """Convert datetime object contained in val1 to jd1, jd2"""
         # Iterate through the datetime objects, getting year, month, etc.
         iterator = np.nditer([val1, None, None, None, None, None, None],
-                             flags=['refs_ok'],
-                             op_dtypes=[object] + 5*[np.intc] + [np.double])
+                             flags=['refs_ok', 'zerosize_ok'],
+                             op_dtypes=[None] + 5*[np.intc] + [np.double])
```

Comment:
What's going on with the change in `op_dtypes` everywhere w/r/t to the `val` type to `None`?  I could not find any docs that suggest what should happen in this case.

Reply:
for inputs, `None` means to just take the dtype from the input. This is needed since for empty arrays, the `dtype` will not be `object` (and there is little point correcting it since no iteration will happen anyway)